pasteit/public/javascripts/create.js

```diff
@@ -1,46 +1,92 @@
+    };
 
-    var languages = [
-              "C++"            , 
-              "C"              , 
-              "Java"           , 
-              "Objective-C"    , 
-              "Clojure"        , 
-              "Apricot"        , 
-              "CoffeeScript"   , 
-              "CSS"            , 
-              "Diff"           , 
-              "Go"             , 
-              "Groovy"         , 
-              "Haskell"        , 
-              "Javascript"     , 
-              "Lua"            , 
-              "Delphi"         , 
-              "Perl"           , 
-              "PHP"            , 
-              "Java Properties", 
-              "Python"         , 
-              "R"              , 
-              "Ruby"           , 
-              "Rust"           , 
-              "Scheme"         , 
-              "Emacs Lisp"     , 
-              "Smalltalk"      , 
-              "Verilog"        , 
-              "XML"            , 
-              "YAML"           , 
-              "HTML"           , 
-              "MySQL"          , 
-              "OCaml"          , 
-              "RPM Spec"        
-    ];
-
-    languages.forEach(function(l) {
-        $("#language").append("<option value=''>"+l+"</option>");
-    });
+    var languages = { 
+        "C++"             : ["clike"],
+        "C"               : ["clike"],
+        "Java"            : ["clike"],
+        "Objective-C"     : ["clike"],
+        "C#"              : ["clike"],
+        "Clojure"         : ["clojure"],
+        "Apricot"         : ["clojure"],
+        "CoffeeScript"    : ["coffeescript"],
+        "CSS"             : ["css"],
+        "Diff"            : ["diff"],
+        "Go"              : ["go"],
+        "Groovy"          : ["groovy"],
+        "Haskell"         : ["haskell"],
+        "Javascript"      : ["javascript"], 
+        "Lua"             : ["lua"],
+        "Delphi"          : ["pascal"],
+        "Perl"            : ["perl"],
+        "PHP"             : ["php"],
+        "Java Properties" : ["properties"],
+        "Python"          : ["python"],
+        "R"               : ["r"],
+        "Ruby"            : ["ruby"],
+        "Rust"            : ["rust"],
+        "Scheme"          : ["scheme"],
+        "Emacs Lisp"      : ["scheme"],
+        "Smalltalk"       : ["smalltalk"],
+        "Verilog"         : ["verilog"],
+        "XML"             : ["xml"],
+        "YAML"            : ["yaml"],
+        "HTML"            : ["htmlmixed"],
+        "MySQL"           : ["mysql"],
+        "OCaml"           : ["ocaml"],
+    };
+
+    var mimes = {
+        "C++"  : "text/x-c++src",
+        "C"    : "text/x-csrc",
+        "C#"   : "text/x-csharp",
+        "Objective-C" : "text/x-objectivec",
+        "Java" : "text/x-java",
+    };
+
+    for(var l in languages) {
+        var h = languages[l];
+        $("#language").append("<option value='"+h+"'>"+l+"</option>");
```

Comment:
Should we look at bringing in a template language like mustache for stuff like this?

Reply:
Let's wait on templating until we find there are many more instances of manually building up inputs.